### PR TITLE
BugFix: DynamoDB Unlock

### PR DESCRIPTION
**Pull request by jconstance-amplify**
Two fixes here... The DynamoDB unlock functionality would fail when
called from repos that are not called terraform-config, so broke out the
function for figuring out the backend state path so it can be reused.

I also added additional error handling where we delete the lock if the
state file is deleted.

### CODE REVIEW

File: bin/tf [hunk 1]
```diff
@@ -163,7 +164,10 @@ def update_digest(error: str, path: str, variables: Dict[str, str]) -> bool:
     :return: True if digest is updated, otherwise False
     """
     dynamodb = DynamoDB(region=variables['region'])
-    digest = re.search(r'[\da-f]{32}', error).group()
+    try:
```
Comment (aoskotsky-amplify) [inline bin/tf]:
```suggestion
    digest = ""
    match = re.search(r'[\da-f]{32}', error)
    if match:
        digest = match.group()
```

File: bin/tf [hunk 2]
```diff
@@ -174,15 +178,22 @@ def update_digest(error: str, path: str, variables: Dict[str, str]) -> bool:
...
+    if digest:
+        response = dynamodb.upsert_item(
+            table_name='terraform-locking',
+            primary_key_name='LockID',
+            primary_key_value=lock_id,
+            attribute_name='Digest',
+            attribute_value=digest
+        )
+    else:
+        response = dynamodb.delete_item(
```
Comment (aoskotsky-amplify) [inline bin/tf]: Why do this?

File: terrawrap/utils/config.py [hunk 3]
```diff
@@ -302,6 +295,24 @@ def calc_backend_config(
     return backend_config
 
 
+def calc_repo_path(path: str) -> str:
```
Comment (aoskotsky-amplify) [inline terrawrap/utils/config.py]: Maybe move this to https://github.com/amplify-education/terrawrap/blob/master/terrawrap/utils/path.py

File: terrawrap/utils/dynamodb.py [hunk 4]
```diff
@@ -32,9 +33,31 @@ def upsert_item(
         expression_attribute_values = {':d': {'S': attribute_value}}
         update_expression = 'SET {} = :d'.format(attribute_name)
 
-        return self.client.update_item(
+        return throttled_call(
```
Comment (aoskotsky-amplify) [inline terrawrap/utils/dynamodb.py]: 👍

File: bin/tf (hunk 2 above)
Comment (jconstance-amplify) [reply bin/tf]: You can end up in a bad state if you delete the S3 state file where the new digest value is blank. Dynamodb complained when I tried to set the digest to empty, so deleting the lockid seems like a reasonable next step.

**Pull request merged**